samtranslator/model/sam_resources.py

```diff
@@ -1785,88 +1787,106 @@ def to_cloudformation(self, **kwargs: Any) -> List[Resource]:  # type: ignore
         resource_resolver: ResourceResolver = kwargs["resource_resolver"]
         original_template = kwargs["original_template"]
 
-        try:
-            destination = get_resource_reference(self.Destination, resource_resolver, self.Source)
-            source = get_resource_reference(self.Source, resource_resolver, self.Destination)
-        except ConnectorResourceError as e:
-            raise InvalidResourceException(self.logical_id, str(e))
-
-        profile = get_profile(source.resource_type, destination.resource_type)
-        if not profile:
-            raise InvalidResourceException(
-                self.logical_id,
-                f"Unable to create connector from {source.resource_type} to {destination.resource_type}; it's not supported or the template is invalid.",
-            )
+        multi_dest = True
+        if isinstance(self.Destination, dict):
+            multi_dest = False
+            self.Destination = [self.Destination]
 
-        # removing duplicate permissions
-        self.Permissions = list(set(self.Permissions))
-        profile_type, profile_properties = profile["Type"], profile["Properties"]
-        profile_permissions = profile_properties["AccessCategories"]
-        valid_permissions_combinations = profile_properties.get("ValidAccessCategories")
+        list_generated_resources: List[Resource] = []
 
-        valid_permissions_str = ", ".join(profile_permissions)
+        for index, dest in enumerate(self.Destination):
```

Comment:
A large nested function is hard to read. Should be possible to separate the concerns into logical parts; adding a 1:1 connector is one thing, and multi-destination is a generalization of that.

Reply:
Synced offline. Will move the majority of for loop into a function `generate_resource`.